src/monitor/group_state_machine.c

```diff
@@ -1031,10 +1085,45 @@ ProceedGroupStateForMSFailover(AutoFailoverNode *activeNode,
 	 */
 	if (nodeBeingPromoted != NULL)
 	{
-		elog(LOG, "Found candidate " NODE_FORMAT,
-			 NODE_FORMAT_ARGS(nodeBeingPromoted));
+		char message[BUFSIZE] = { 0 };
+
+		List *knownUnreachableStates =
+			list_make2_int(REPLICATION_STATE_REPORT_LSN,
+						   REPLICATION_STATE_PREPARE_PROMOTION);
+
+		/* activeNode might be the failover candidate, proceed already */
+		if (nodeBeingPromoted->nodeId == activeNode->nodeId)
+		{
+			return ProceedWithMSFailover(activeNode, nodeBeingPromoted);
+		}
 
-		return ProceedWithMSFailover(activeNode, nodeBeingPromoted);
+		LogAndNotifyMessage(
+			message, BUFSIZE,
+			"Active " NODE_FORMAT
+			" found failover candidate " NODE_FORMAT
+			" being promoted, it reported state \"%s\"",
+			NODE_FORMAT_ARGS(activeNode),
+			NODE_FORMAT_ARGS(nodeBeingPromoted),
+			ReplicationStateGetName(nodeBeingPromoted->reportedState));
+
+		/*
+		 *
```

Comment:
There was an extra newline in this comment.

```suggestion
```

Reply:
Ooops, sorry about that. I guess that's one way to check the stability of the test suite.